base/multidimensional.jl

```diff
@@ -249,8 +249,9 @@ module IteratorsMD
         convert(Tuple{Vararg{UnitRange{Int}}}, R)
 
     # AbstractArray implementation
+    Base.axes(iter::CartesianIndices{N,R}) where {N,R} = iter.indices
     Base.IndexStyle(::Type{CartesianIndices{N,R}}) where {N,R} = IndexCartesian()
-    @inline Base.getindex(iter::CartesianIndices{N,R}, I::Vararg{Int, N}) where {N,R} = CartesianIndex(first.(iter.indices) .- 1 .+ I)
+    @inline Base.getindex(iter::CartesianIndices{N,R}, I::Vararg{Int, N}) where {N,R} = CartesianIndex(I)
```

Comment:
Actually, on second thought — we need to add `@boundscheck checkbounds(iter, I)` to this implementation, too.  Now's the time to do it since this is definitely a performance gain right now.

Reply:
Makes sense. I've added a commit (AFAICT that's `checkbounds(iter, I...)`).